rune/libenclave/intelsgx/arch.go

```diff
@@ -128,26 +129,29 @@ type Report struct {
 }
 
 type ReportBody struct {
-	CpuSvn       [16]byte `struct:"[16]byte"`
-	MiscSelect   uint32   `struct:"uint32"`
-	_            [12]byte `struct:"[12]byte"`
+	CpuSvn     [16]byte `struct:"[16]byte"`
+	MiscSelect uint32   `struct:"uint32"`
+	_          [12]byte `struct:"[12]byte"`
+	// IsvExtProdId is reserved in Quote V3
 	IsvExtProdId [16]byte `struct:"[16]byte"`
 	Attributes   [16]byte `struct:"[16]byte"`
 	MrEnclave    [32]byte `struct:"[32]byte"`
 	_            [32]byte `struct:"[32]byte"`
 	MrSigner     [32]byte `struct:"[32]byte"`
 	_            [32]byte `struct:"[32]byte"`
-	ConfigId     [64]byte `struct:"[64]byte"`
-	IsvProdId    uint16   `struct:"uint16"`
-	IsvSvn       uint16   `struct:"uint16"`
-	ConfigSvn    uint16   `struct:"uint16"`
-	_            [42]byte `struct:"[42]byte"`
-	IsvFamilyId  [16]byte `struct:"[16]byte"`
-	ReportData   [64]byte `struct:"[64]byte"`
+	// ConfigId is reserved in Quote V3
```

Comment:
ditto

Reply:
Please add "signature[]" into quote structure as below:
typedef struct _quote_t
{
    uint16_t            version;        /* 0   */
    uint16_t            sign_type;      /* 2   */
    sgx_epid_group_id_t epid_group_id;  /* 4   */
    sgx_isv_svn_t       qe_svn;         /* 8   */
    sgx_isv_svn_t       pce_svn;        /* 10  */
    uint32_t            xeid;           /* 12  */
    sgx_basename_t      basename;       /* 16  */
    sgx_report_body_t   report_body;    /* 48  */
    uint32_t            signature_len;  /* 432 */
    uint8_t             signature[];    /* 436 */
} sgx_quote_t;


typedef struct _sgx_quote3_t {
    sgx_quote_header_t  header;              ///< 0:   The quote header.
    sgx_report_body_t   report_body;         ///< 48: The REPORT of the app that is attesting remotely.
    uint32_t            signature_data_len;  ///< 432: The length of the signature_data.  Varies depending on the type of sign_type.
#ifdef _MSC_VER
#pragma warning(push)
#pragma warning ( disable:4200 )
#endif
    uint8_t             signature_data[];    ///< 436: Contains the variable length containing the quote signature and support data for the signature.
#ifdef _MSC_VER
#pragma warning(pop)
#endif
} sgx_quote3_t;


You can see whatever quote or quotes, they have the same content in the first 436 bytes.

Reply:
Please see two quote header, you can define them in as sgx_quote_header_t and sgx_quote3_header_t
quote header：
{
uint16_t version; /* 0 /
uint16_t sign_type; / 2 /
sgx_epid_group_id_t epid_group_id; / 4 /
sgx_isv_svn_t qe_svn; / 8 /
sgx_isv_svn_t pce_svn; / 10 /
uint32_t xeid; / 12 /
sgx_basename_t basename; / 16 /
}
quote3 header：
typedef struct _sgx_quote3_header_t {
    uint16_t            version;             ///< 0:  The version this quote structure.
    uint16_t            att_key_type;        ///< 2:  sgx_attestation_algorithm_id_t.  Describes the type of signature in the signature_data[] field.
    uint32_t            att_key_data_0;      ///< 4:  Optionally stores additional data associated with the att_key_type.
    sgx_isv_svn_t       qe_svn;              ///< 8:  The ISV_SVN of the Quoting Enclave when the quote was generated.
    sgx_isv_svn_t       pce_svn;             ///< 10: The ISV_SVN of the PCE when the quote was generated.
    uint8_t             vendor_id[16];       ///< 12: Unique identifier of QE Vendor.
    uint8_t             user_data[20];       ///< 28: Custom attestation key owner data.
} sgx_quote3_header_t;

Reply:
> Please see two quote header, you can define them in as sgx_quote_header_t and sgx_quote3_header_t
> quote header：
> {
> uint16_t version; /* 0 /
> uint16_t sign_type; / 2 /
> sgx_epid_group_id_t epid_group_id; / 4 /
> sgx_isv_svn_t qe_svn; / 8 /
> sgx_isv_svn_t pce_svn; / 10 /
> uint32_t xeid; / 12 /
> sgx_basename_t basename; / 16 /
> }
> quote3 header：
> typedef struct _sgx_quote3_header_t {
> uint16_t version; ///< 0: The version this quote structure.
> uint16_t att_key_type; ///< 2: sgx_attestation_algorithm_id_t. Describes the type of signature in the signature_data[] field.
> uint32_t att_key_data_0; ///< 4: Optionally stores additional data associated with the att_key_type.
> sgx_isv_svn_t qe_svn; ///< 8: The ISV_SVN of the Quoting Enclave when the quote was generated.
> sgx_isv_svn_t pce_svn; ///< 10: The ISV_SVN of the PCE when the quote was generated.
> uint8_t vendor_id[16]; ///< 12: Unique identifier of QE Vendor.
> uint8_t user_data[20]; ///< 28: Custom attestation key owner data.
> } sgx_quote3_header_t;

Sure, I already achieve this in `arch.go` in the pr.

Reply:
> Please add "signature[]" into quote structure as below:
> typedef struct _quote_t
> {
> uint16_t version; /* 0 _/ uint16_t sign_type; /_ 2 _/ sgx_epid_group_id_t epid_group_id; /_ 4 _/ sgx_isv_svn_t qe_svn; /_ 8 _/ sgx_isv_svn_t pce_svn; /_ 10 _/ uint32_t xeid; /_ 12 _/ sgx_basename_t basename; /_ 16 _/ sgx_report_body_t report_body; /_ 48 _/ uint32_t signature_len; /_ 432 _/ uint8_t signature[]; /_ 436 */
> } sgx_quote_t;
> 
> typedef struct _sgx_quote3_t {
> sgx_quote_header_t header; ///< 0: The quote header.
> sgx_report_body_t report_body; ///< 48: The REPORT of the app that is attesting remotely.
> uint32_t signature_data_len; ///< 432: The length of the signature_data. Varies depending on the type of sign_type.
> #ifdef _MSC_VER
> #pragma warning(push)
> #pragma warning ( disable:4200 )
> #endif
> uint8_t signature_data[]; ///< 436: Contains the variable length containing the quote signature and support data for the signature.
> #ifdef _MSC_VER
> #pragma warning(pop)
> #endif
> } sgx_quote3_t;
> 
> You can see whatever quote or quote3, they have the same content in the first 436 bytes.

Sure, I already achieve this in `arch.go` in the pr.